python/ray/experimental/worker.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import ray
+import numpy as np
+
+
+def get(object_ids, worker=None):
+    """Get a remote object or a collection of remote objects from the object
+    store.
+
+    This method blocks until the object corresponding to the object ID is
+    available in the local object store. If this object is not in the local
+    object store, it will be shipped from an object store that has it (once the
+    object has been created). If object_ids is a list, tuple or numpy array,
+    then the objects corresponding to each object in the list will be returned.
+    If object_ids is a dict, then the objects corresponding to the ObjectIDs in
+    the dict's values will be gotten from the object store and the dict will
+    return.
+
+    Args:
+        object_ids: Object ID of the object to get, a list, tuple, ndarray of
+            object IDs to get or a dict of {key: object ID}.
+
+    Returns:
+        A Python object, a list of Python objects or a dict of {key: object}.
+    """
+    if isinstance(object_ids, (list, tuple, np.ndarray)):
+        # There is a dependency on ray.worker which prevents importing
+        # global_worker at the top of this file
+        return ray.get(list(object_ids)) if worker is None else \
+            ray.get(list(object_ids), worker)
+    elif isinstance(object_ids, dict):
+        to_get = [
```

Comment:
I think this can be simplified, maybe something like

```python
keys_to_get = [k for k, v in object_ids.items() if isinstance(v, ray.ObjectID)]
ids_to_get = [v for k, v in object_ids.items() if isinstance(v, ray.ObjectID)]
values = ray.get(ids_to_get)

result  = object_ids.copy()
for key, value in zip(keys_to_get, values):
    result[key] = value
```